backend/write-transaction-impl.ts

```diff
@@ -0,0 +1,68 @@
+import type { JSONValue, ScanResult, WriteTransaction } from "replicache";
+import { delObject, getObject, putObject } from "./data";
+import { ExecuteStatementFn, transact } from "./rds";
+
+/**
+ * Implements ReplicaCache's WriteTransaction interface in terms of a MySQL
+ * transaction.
+ */
+export class WriteTransactionImpl implements WriteTransaction {
+  private _docID: string;
+  private _executor: ExecuteStatementFn;
+  private _cache: Record<
+    string,
+    { value: JSONValue | undefined; dirty: boolean }
+  > = {};
```

Comment:
General rule is to use Maps for maps.

Otherwise you run into issues with keys like `toString` etc